src/DssSpell.sol

```diff
@@ -49,81 +73,328 @@ contract DssSpellAction is DssAction {
 
     // ---------- Math ----------
     uint256 internal constant WAD = 10 ** 18;
+    uint256 internal constant RAY = 10 ** 27;
 
     // ---------- Contracts ----------
-    address internal constant USDS_OFT        = 0x1e1D42781FC170EF9da004Fb735f56F0276d01B8;
-    // Note: base58 BEvTHkTyXooyaJzP8egDUC7WQK8cyRrq5WvERZNWhuah to hex conversion can be checked at https://emn178.github.io/online-tools/base58/decode/?input=BEvTHkTyXooyaJzP8egDUC7WQK8cyRrq5WvERZNWhuah&output_type=hex
-    bytes32 internal constant SOLANA_USDS_OFT = 0x9825dc0cbeaf22836931c00cb891592f0a96d0dc6a65a4c67992b01e0db8d122;
-    address internal constant LZ_GOV_SENDER   = 0x27FC1DD771817b53bE48Dc28789533BEa53C9CCA;
-    // Note: base58 8vXXGiaXFrKFUDw21H5Z57ex552Lh8WP9rVd2ktzmcCy to hex conversion can be checked at https://emn178.github.io/online-tools/base58/decode/?input=8vXXGiaXFrKFUDw21H5Z57ex552Lh8WP9rVd2ktzmcCy&output_type=hex
-    bytes32 internal constant SOLANA_LZ_GOV   = 0x75b81a4430dee7012ff31d58540835ccc89a18d1fc0522bc95df16ecd50efc32;
-    address internal constant LZ_GOV_RELAY    = 0x2beBFe397D497b66cB14461cB6ee467b4C3B7D61;
+    address internal immutable CHAINLOG                 = DssExecLib.LOG;
+    address internal immutable DAI                      = DssExecLib.dai();
+    address internal immutable MCD_VAT                  = DssExecLib.vat();
+    address internal immutable MCD_JUG                  = DssExecLib.jug();
+    address internal immutable MCD_VOW                  = DssExecLib.vow();
+    address internal immutable MCD_SPBEAM               = DssExecLib.getChangelogAddress("MCD_SPBEAM");
+    address internal immutable ALLOCATOR_SPARK_A_VAULT  = DssExecLib.getChangelogAddress("ALLOCATOR_SPARK_A_VAULT");
+    address internal immutable ALLOCATOR_BLOOM_A_VAULT  = DssExecLib.getChangelogAddress("ALLOCATOR_BLOOM_A_VAULT");
+    address internal immutable SKY                      = DssExecLib.getChangelogAddress("SKY");
+    address internal immutable DAI_USDS                 = DssExecLib.getChangelogAddress("DAI_USDS");
+    address internal immutable LZ_GOV_SENDER            = DssExecLib.getChangelogAddress("LZ_GOV_SENDER");
+    address internal immutable CRON_STARGUARD_JOB       = DssExecLib.getChangelogAddress("CRON_STARGUARD_JOB");
+
+    address internal constant GROVE_STARGUARD       = 0xfc51CAa049E8894bEcFfB68c61095C3F3Ec8a880;
+    address internal constant KEEL_STARGUARD        = 0xe8fF70481d653Ec31AB9E0cB2A8B316afF8D84ee;
+    address internal constant OBEX_STARGUARD        = 0x987f1C31f9935e9926555BcFB76516bb2EcEccaD;
+
+    // ---------- Spark Spell ----------
+    address internal immutable SPARK_STARGUARD  = DssExecLib.getChangelogAddress("SPARK_STARGUARD");
+    address internal constant  SPARK_SPELL      = 0x2C9E477313EC440fe4Ab6C98529da2793e6890F2;
```

Comment:
Confirming spell address is correct: `0x2C9E477313EC440fe4Ab6C98529da2793e6890F2`